ASM/c/models.c

```diff
@@ -127,7 +127,7 @@ void lookup_model_by_override(model_t *model, override_t override) {
         uint16_t item_id = override.value.looks_like_item_id ?
             override.value.looks_like_item_id :
             override.value.base.item_id;
-        uint16_t resolved_item_id = resolve_upgrades(item_id);
+        uint16_t resolved_item_id = resolve_upgrades(override);
```

Comment:
`item_id` needs to refactored to an override to pass to `resolve_upgrades`. This is the cause of the ice trap freestanding item failures. Right now it's passing the ice trap ID instead of the looks-like item ID. Ice traps have null model IDs and graphic IDs, which causes the crash on draw.

My suggestion uses a temp variable to prevent mutating the original override, but I don't know how important that is. Better safe than sorry IMO.

```suggestion
        override_t model_override = override;
        model_override.value.base.item_id = override.value.looks_like_item_id ?
            override.value.looks_like_item_id :
            override.value.base.item_id;
        uint16_t resolved_item_id = resolve_upgrades(model_override);
```

Reply:
I've tested this on my fork (which is based on 7.1.118) and it works. For rebasing onto latest Dev, it looks like `lookup_shop_draw_id` needs a similar adjustment.